torch_xla/experimental/pjrt.py

```diff
@@ -220,6 +233,49 @@ def _discover_tpu_master_worker_ip(device: torch.device):
   return _merge_replica_results(replica_results)
 
 
+@requires_pjrt
+def _run_singleprocess(fn: Callable[..., R],
+                       *args,
+                       start_method: str = 'spawn',
+                       master_port: int = 12355,
+                       **kwargs) -> Dict[int, R]:
+  """Runs `fn` on a single device core.
+
+  Spawns one process on a single physical device (e.g. TPU chip).
+
+  Args:
+    fn: Function to run on the device devices
+    args: args to pass to `fn`
+    start_method: The Python `multiprocessing` process creation method.
+      Default: `spawn`
+    kwargs: kwargs to pass to `fn`
+
+  Returns:
+    the result of calling `fn`.
+  """
+  with concurrent.futures.ProcessPoolExecutor(
```

Comment:
This outer `executor` isn't used at all

I think that's actually okay though. The implementation that we're mirroring also does not create any child processes for a single-device run.

https://github.com/pytorch/xla/blob/a1c054dadfa44aa320fe58b73883ad3c23c0912a/torch_xla/distributed/xla_multiprocessing.py#L397-L398

So I think you can just remove this context and follow the logic of `xmp.spawn`. The main reason we would want to create a child process is so any state (especially TPU topology env vars and the TPU client itself) we alter while running `fn` effectively gets reset when the child exits. I guess a user might also reasonably expect `xmp.spawn` to _spawn_ something.

@JackCaoG what do you think is the better option?

Reply:
I don't see a need to create child process for a single-device. If this is significantly easy to implement this way I am also ok with it but it does seems a bit redundant.

Reply:
Sounds good. @stgpetrovic can you remove both executors?